test/smoke_test/bcda_client.go

```diff
@@ -84,6 +85,26 @@ func main() {
 	}
 }
 
+func NewClient(accessToken string, retries int) *client {
+	c := &client{accessToken: accessToken}
+
+	retryClient := retryablehttp.NewClient()
+	retryClient.RetryMax = retries
+	retryClient.CheckRetry = func(ctx context.Context, resp *http.Response, err error) (bool, error) {
+		if resp.StatusCode == http.StatusUnauthorized {
+			log.Info("Access token expired. Refreshing...")
```

Comment:
De we need to add:
`	
if ctx.Err() != nil {
		return false, ctx.Err()
	}
`
here?

Reply:
Sorry. Add it after line 93.

Reply:
No, on line 102 I call the `DefaultRetryPolicy`. It actually does all the error checking we need.

Reply:
DefaultRetryPolicy handles ctx, but is it handled when StatusCode == StatusUnathorized?

Reply:
It will retry all but three errors and will retry all 5xx (except 501). It does not, by default, retry a `401`, hence me adding in the `StatusUnauthorized` check.